design/DFX-Interface.md

```diff
@@ -99,22 +91,27 @@ This file informs Motoko about the interface of that canister. It could be the o
 
 Open problem: how to resolve mutual canister imports.
 
-Compiling IDL Files to JS
--------------------------
-
-In order to compile an IDL file, `dfx` invokes `didc` with
-
-    didc --js some/path/input.did -o another/path/output.js
-
-This _reads_ `some/path/input.did` and any `.did` file referenced by
-`some/path/input.did`.
-
-No constraints are imposed where these imported files reside (this may be refined to prevent relative imports from looking outside the project and the declared packages)
-
-This _writes_ to `another/path/output.js`, but has no other effect. It does
-not create `another/path/`.
-
-Compiler warnings and errors are reported to `stderr`. Nothing writes to `stdout`.
+Exporting Canister Metadata
+---------------------------
+
+The compiler generates various metadata about the canister via command line flags.
+The compiled Wasm module also includes the metadata in the custom sections.
+
+* Candid interface.
+  + Compiler flag `--idl` generates the Candid interface for the canister. The main service
```

Comment:
Given that the ecosystem (e.g. upgrades) will depend on this, I wonder if this shouldn't become the default when compiling for the IC, and we'd rather have a flag `--no-idl` (and maybe `--private-idl`, see below).

Reply:
`--idl` flag only outputs the candid interface to a file. The normal compilation process will always include the candid interface in the custom section. We add the `--public-metadata` flag to control if the section should be public or private.